lib/ngx/balancer.lua

```diff
@@ -135,10 +180,23 @@ function _M.set_timeouts(connect_timeout, send_timeout, read_timeout)
         read_timeout = read_timeout * 1000
     end
 
-    local rc =
-        C.ngx_http_lua_ffi_balancer_set_timeouts(r, connect_timeout,
-                                                 send_timeout, read_timeout,
-                                                 errmsg)
+    local rc
+
+    if subsystem == 'http' then
+        rc = C.ngx_http_lua_ffi_balancer_set_timeouts(r, connect_timeout,
+                                                      send_timeout,
+                                                      read_timeout,
+                                                      errmsg)
+
+    elseif subsystem == 'stream' then
+        local timeout = max(send_timeout, read_timeout)
+
+        rc = C.ngx_stream_lua_ffi_balancer_set_timeouts(r, connect_timeout,
+                                                        timeout, errmsg)
+
+    else
+        error("unknown subsystem: " .. subsystem)
+    end
```

Comment:
I don't like these branches in such a hot code path. Can we add an `ngx_lua_ffi_balancer_set_timeouts` function which is in the same spirit as the other dynamically dispatched helpers?

Reply:
Problem is that those two functions have different signature since the stream version does not have as much timeout values as the http version.